src/index.js

```diff
@@ -40,13 +46,40 @@ module.exports = {
     debug(`Booting device (${device.name} ${device.os} ${device.version})`);
     // Timeout in seconds
     const timeout = process.env.IOS_BOOT_TIMEOUT || 60;
-
     await idbCompanion.boot(device.udid, timeout * 1000);
 
-    debug(`Opening url: ${pageUrl}`);
-    await exec(`xcrun simctl openurl ${device.udid} ${pageUrl}`, {
-      stdio: "ignore",
-    });
+    const maxRetries = 5;
+    const retryDelay = 2000;
+    let attempt = 0;
+
+    while (attempt < maxRetries) {
+      attempt++;
+
+      try {
+        // Try opening the URL on the device by udid
+        await exec(`xcrun simctl openurl ${device.udid} ${pageUrl}`);
+        return; // Success, exit function.
+      } catch (error) {
+        debug(`Error opening URL: ${error}`);
+        if (attempt >= maxRetries) {
+          throw new Error(
+            `Failed to open URL on simulator after ${maxRetries} attempts.`,
+          );
```

Comment:
The original behavior was to throw the original exception. Perhaps we should keep this?

Reply:
Agreed, good catch. I have updated the throw to include the error message.

Reply:
I'm thinking, maybe we should use the original error on these, and update just the message? It'll make it easier to find the place where the error was thrown.


```suggestion
          if (error instanceof Error) {
            error.message = `Failed to open URL on simulator after ${maxRetries} attempts. Last error: ${error.message}`;
            throw error;
          }
          throw new Error(
            `Failed to open URL on simulator after ${maxRetries} attempts. Last error: ${String(error)}`,
          );
```